javalib/src/main/scala/java/util/concurrent/ConcurrentSkipListSet.scala

```diff
@@ -0,0 +1,423 @@
+
+  def lower(e: E): E =
+    headSet(e, false).lastOption.getOrElse(null.asInstanceOf[E])
+
+  def pollFirst(): E = {
+    val polled = inner.headOption
+    if (polled.isDefined) {
+      val elem = polled.get.inner
+      remove(elem)
+      elem
+    } else null.asInstanceOf[E]
+  }
+
+  def pollLast(): E = {
+    val polled = inner.lastOption
+    if (polled.isDefined) {
+      val elem = polled.get.inner
+      remove(elem)
+      elem
+    } else null.asInstanceOf[E]
+  }
+
+  override def remove(o: Any): Boolean =
+    if (o == null) throw new NullPointerException()
+    else inner.remove(Box(o.asInstanceOf[E]))
+
+  override def removeAll(c: Collection[_]): Boolean = {
+    val iter = c.iterator()
+    var changed = false
+    while (iter.hasNext)
+      changed = inner.remove(iter.next) || changed
+
+    changed
+  }
+
+  def size(): Int =
+    inner.size
+
+  def headSet(toElement: E): NavigableSet[E] =
+    headSet(toElement, false)
+
+  def headSet(toElement: E, inclusive: Boolean): NavigableSet[E] = {
+    val headSetFun = () => {
+      val boxed = Box(toElement)
+      val hs = new mutable.TreeSet[Box[E]]()
+      hs ++= inner.until(boxed)
+      if (inclusive && inner.contains(boxed))
+        hs += boxed
+      hs
+    }
+    new ConcurrentSkipListSet.NavigableView(this, headSetFun)
+  }
+
+  def tailSet(fromElement: E): NavigableSet[E] =
+    tailSet(fromElement, true)
+
+  def tailSet(fromElement: E, inclusive: Boolean): NavigableSet[E] =  {
+    val tailSetFun = () => {
+      val boxed = Box(fromElement)
+      val ts = new mutable.TreeSet[Box[E]]()
+      ts ++= inner.from(boxed)
+      if (!inclusive && inner.contains(boxed))
+        ts remove boxed
+      ts
+    }
+    new ConcurrentSkipListSet.NavigableView(this, tailSetFun)
+  }
+
+  def subSet(fromElement: E, fromInclusive: Boolean, toElement: E, toInclusive: Boolean): NavigableSet[E] = {
+    val subSetFun = () => {
+      val boxedFrom = Box(fromElement)
+      val boxedTo = Box(toElement)
+      val ss = new mutable.TreeSet[Box[E]]()
+      ss ++= inner.range(boxedFrom, boxedTo)
+      if (!fromInclusive && inner.contains(boxedFrom))
+        ss remove boxedFrom
+      if (toInclusive && inner.contains(boxedTo))
+        ss += boxedTo
+      ss
+    }
+    new ConcurrentSkipListSet.NavigableView(this, subSetFun)
+  }
+
+  def subSet(fromElement: E, toElement: E): NavigableSet[E] =
+    subSet(fromElement, true, toElement, false)
+}
+
+object ConcurrentSkipListSet {
+
+  private def defaultOrdering[E]: Ordering[E] = {
+    new Ordering[E] {
+      def compare(a: E, b: E): Int =
+        a.asInstanceOf[Comparable[E]] compareTo b
+    }
+  }
+
+  private class NavigableView[E](original: NavigableSet[E],
+                         inner: () => mutable.SortedSet[Box[E]])
+                            extends AbstractCollection[E]
+                            with NavigableSet[E]
+                            with SortedSet[E] {
```

Comment:
Indentation issues here. All these lines should be indented 4 spaces wrt `private`, so they should start with 6 spaces total. This also leaves enough room to put the `with`s on the same line as the `extends`.